### PR TITLE
[wip] Fix update ability, bump api version, minor fixes

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Update or delete outdated comments
- Windows installer setting name

### CODE REVIEW

File: CorsixTH/Lua/humanoid_actions/die.lua [hunk 1]
```diff
@@ -261,8 +261,6 @@ local function action_die_start(action, humanoid)
   local anims = humanoid.die_anims
   assert(anims, "Error: no death animation for humanoid ".. humanoid.humanoid_class)
   action.must_happen = true
-  -- TODO: Right now the angel version of death is the only possibility
-  -- The Grim Reaper should sometimes also have a go.
   local fall = anims.fall_east
 
   --If this isn't done their bald head will become bloated instead of suddenly having hair:
```
Comment (lewri) [inline CorsixTH/Lua/humanoid_actions/die.lua:266]: This animation is recorded in #495 / #1638 but there appears to be code now.

Comment (tobylane) [reply CorsixTH/Lua/humanoid_actions/die.lua:266]: The todo is from 11 years ago and is resolved, the Reaper is implemented. Can't speak to those issues.

Comment (lewri) [reply CorsixTH/Lua/humanoid_actions/die.lua:266]: Sorry I meant the comment at line 266.

File: CorsixTH/Lua/dialogs/resizables/update.lua [hunk 2]
```diff
@@ -109,8 +101,10 @@ end
 
 function UIUpdate:buttonDownload()
 
-  if self.os_is_windows then
+  if self.app.os == "windows" then
     os.execute("start " .. self.download_url)
```
Comment (Alberth289346) [inline CorsixTH/Lua/dialogs/resizables/update.lua:105]: Not your fault, but this is really a good way to get your system hacked.

**Comment by TheCycoONE** [comment]
Besides the conflicts, what is the state of this?

**Comment by tobylane** [comment]
I'll break it up into smaller PRs.

**Pull request closed without merging**